src/android/CodePush.java

```diff
@@ -461,6 +487,8 @@ public void onStart() {
                 handleAppStart();
                 this.markUpdate();
                 this.codePushPackageManager.clearPendingInstall();
+            } else if (codePushReportingManager.hasFailedReport()) {
```

Comment:
We want to check for failed reports everytime the app resumes, not just when there's an update.